literals/literals.go

```diff
@@ -1,15 +1,147 @@
-package main
+package literals
 
 import (
 	"fmt"
 	"go/ast"
 	"go/token"
 	"go/types"
+	"log"
+	mathrand "math/rand"
 	"strconv"
 
 	"golang.org/x/tools/go/ast/astutil"
 )
 
+type stringObfuscator interface {
+	obfuscateString(data string) *ast.BlockStmt
+}
+
+type boolObfuscator interface {
+	obfuscateBool(data bool) *ast.BlockStmt
+}
+
+type byteObfuscator interface {
+	obfuscateByte(data []byte) *ast.BlockStmt
+}
+
+type intObfuscator interface {
+	obfuscateInt(data int) *ast.BlockStmt
```

Comment:
`int` is not enough to obfuscate any integer. what if you're on 32-bit and want to obfuscate an uint32, or an int64?

you could use int64, but still, that isn't going to work with uint64, and vice versa.